sdks/go/pkg/beam/io/synthetic/source.go

```diff
@@ -235,6 +247,12 @@ func (b *SourceConfigBuilder) Build() SourceConfig {
 	if b.cfg.ValueSize <= 0 {
 		panic(fmt.Sprintf("SourceConfig.ValueSize must be >= 1. Got: %v", b.cfg.ValueSize))
 	}
+	if b.cfg.HotKeys <= 0 {
+		panic(fmt.Sprintf("SourceConfig.HotKeys must be >= 1. Got: %v", b.cfg.HotKeys))
+	}
+	if b.cfg.HotKeyFraction <= 0 {
+		panic(fmt.Sprintf("SourceConfig.HotKeyFraction must be >= 1. Got: %v", b.cfg.HotKeyFraction))
```

Comment:
HotKeyFraction is supposed to be a floating point number from 0 to 1, right?

Reply:
```
	if b.cfg.HotKeyFraction < 0 || b.cfg.HotKeyFraction > 1 {
		panic(fmt.Sprintf("SourceConfig.HotKeyFraction must be a floating point number from 0 and 1. Got: %v", b.cfg.NumHotKeys))
	}
```